util/dtgen/helper.py

```diff
@@ -547,6 +616,28 @@ def _init_clocks(self):
         if isinstance(self.reg_block_enum, CEnum):
             self.clock_enum.add_constant(Name(["count"]), "Number of clock ports")
 
+    def has_reset_requests(self):
+        return len(self.reset_req_map) > 0
+
+    def _init_reset_requests(self):
+        self.reset_req_enum = self._enum_type(Name([]), Name(["dt"]) + self.ip_name +
+                                              Name(["reset", "req"]))
+        self.reset_req_map = OrderedDict()
```

Comment:
Random aside: a "reset request" network seems like an atypical thing that only OT does, hehe.

Reply:
Maybe because resets are more carefully handled? There are several reset requests in other systems, POR, magic reset key presses, watchdogs...